core/templates/dev/head/pages/exploration_editor/editor_tab/SolutionVerificationService.js

```diff
@@ -30,6 +32,9 @@ oppia.factory('SolutionVerificationService', [
           AnswerClassificationService.getMatchingClassificationResult(
             stateName, interaction, correctAnswer, rulesService
           ));
+        if (StateEditorService.getInQuestionMode()) {
```

Comment:
Probably should have a Karma test for this

Reply:
Done, added.